### PR TITLE
First login is automated + errors are displayed on the login page

### PR DESCRIPTION
**Pull request by w10d**
Change-Id: Iddfcfd8206a69fe3bae22daddcab291699d89872

### CODE REVIEW

File: bundles/org.eclipse.orion.server.authentication/src/org/eclipse/orion/server/authentication/oauth/OAuthHelper.java [hunk 1]
```diff
@@ -140,10 +140,13 @@ public static void handleLogin(HttpServletRequest req, HttpServletResponse resp,
 			if (!FormAuthHelper.canAddUsers()) {
 				throw new OAuthException("There is no Orion account associated with this Id. Please register or contact your system administrator for assistance.");
 			}
-			String url = "/mixloginstatic/LoginWindow.html";
+			String url = "/mixloginstatic/register.html";
```
Comment (aaudiber) [inline bundles/org.eclipse.orion.server.authentication/src/org/eclipse/orion/server/authentication/oauth/OAuthHelper.java]: extra space at end

**Comment by aaudiber** [comment]
url += "&oauth-redirect=true"; 

What does this query param do? I don't see oauth-redirect anywhere else in the codebase.

**Comment by w10d** [comment]
> url += "&oauth-redirect=true";

check the client PR. it is necessary in order to distinguish between the regular oauth (like google) and the oauth with redirection

**Comment by aaudiber** [comment]
I see. All forms of OAuth use redirection though, so we should improve the name to something like "skip-registration-page"

**Pull request merged**